File: app/src/main/cpp/counter.cpp
//
// Created by hiroki on 2020/03/31.
//

#include "counter.hpp"

counter::counter() {
    count = 0;
}

string counter::description() {
    return std::to_string(count) + "回実行されました";
}

void counter::increment() {
    count++;
}

File: app/src/main/java/com/example/cppclasstest/Counter.kt
package com.example.cppclasstest

class Counter {

    private var pointer: Long = 0 // C++インスタンスのポインタを保持するフィールド

    var count: Int
    get() = getCountCpp()
    set(value) = setCountCpp(value)
    private external fun getCountCpp(): Int
    private external fun setCountCpp(value: Int)

    external fun description(): String
    external fun increment()

    external fun delete()
    protected fun finalize() = delete()
}